pkg/controller/controller_instance.go

```diff
@@ -89,8 +91,22 @@ const (
 	startingInstanceOrphanMitigationMessage string = "The instance provision call failed with an ambiguous error; attempting to deprovision the instance in order to mitigate an orphaned resource"
 
 	clusterIdentifierKey string = "clusterid"
+
+	minBrokerProvisioningRetryDelay time.Duration = time.Second * 1
+	maxBrokerProvisioningRetryDelay time.Duration = time.Minute * 20
 )
 
+type retryQueue struct {
```

Comment:
this strikes me as the wrong place to define this, or it needs a better name, as it seems very provision+instance specific.

Reply:
granted I'm not familiar with the existing queues, but is there a reason to avoid using some version of  "k8s.io/client-go/util/workqueue" ?

Reply:
This is very instance & provision oriented.  This isn't meant to be long term - it's a long read, but #2076 discusses many of the existing issues we're seeing in here.  The instance controller really needs a major refactoring - I don't picture this new provisionRetryQueue code being part of it.  This is purely to plug a hole specific instance provisioning retry.

Reply:
okay, thanks